proto/protovalidate/buf/validate/validate.proto

```diff
@@ -4770,8 +4770,19 @@ message Violations {
 // }
 // ```
 message Violation {
-  // `field_path` is a machine-readable identifier that points to the specific field that failed the validation.
+  // `field` is a machine-readable path to the field that failed validation.
   // This could be a nested field, in which case the path will include all the parent fields leading to the actual field that caused the violation.
+  optional FieldPath field = 5;
+
+  // `rule` is a machine-readable path that points to the specific constraint rule that failed validation.
+  // This will be a nested field starting from the FieldConstraints of the field that failed validation.
+  // This field is not present when there is no corresponding rule field, e.g. for custom constraints.
+  optional FieldPath rule = 6;
```

Comment:
For both of these, can you add an in-line example to demonstrate effectively what these would be? If that's a bit too much to fit in the proto, can we add a doc that talks about this and gives examples?

Reply:
Doesn't seem too bad to just add a simple example for each. Done.